jdaviz/configs/imviz/plugins/coords_info/coords_info.py

```diff
@@ -407,9 +407,13 @@ def _image_viewer_update(self, viewer, x, y):
         # Extract data values at this position.
         # Check if shape is [x, y, z] or [y, x] and show value accordingly.
         if image.ndim == 3:
-            # needed for cubeviz
-            ix_shape = 0
-            iy_shape = 1
+            if self.app.config == 'lcviz':
```

Comment:
Is there no other way? I don't know how I feel about having special logic for a config that does not exist in this repo. This might be a slippery slope.

Reply:
see my comment - if you can think of a way to generalize that line if these indices are stored in the helper, then I am happy to do that.  The other option would be to refactor so that this part can be overridden without having to copy and paste the rest of the entire method (perhaps a `_get_image_layer_data` method).

Reply:
Even if we manage to generalize this, what about things like this?

```python
    } else if (this.$props.viewer.config === 'lcviz') {
      if (this.$props.viewer.reference.startsWith('tpf')) {
        multi_select = false
      }
```

Reply:
we have a ticket to refactor that to be more downstream-friendly.  Until we can prioritize that, at least these will live in dedicated if-statements 🤷‍♂️

Reply:
I think if you do `if config == "cubeviz"` instead, then we don't have to worry about lcviz logic sneaking in here as lcviz then would go the route of "else".

Reply:
It just felt to me like that would be more obfuscated and harder to remember to remove when we do the refactor

Reply:
I still feel better if we don't hardcode "lcviz" in Imviz plugin.

Reply:
Coords info is now used in all configs, perhaps we should move it to a more general location then.

Either way, I can do the refactor for this part now if you'd prefer? The vue file is a much more involved effort than will need to wait until it can be prioritized.

Reply:
If you swap out `if config == "lcviz"` with `if config == "cubeviz"`, the functionality stays the same and I will stop bugging you. 😆

Reply:
I'd rather refactor - I don't like the idea of having the fallback behavior in jdaviz be the behavior needed by lcviz.  That will make it too hard to remove the logic when we do do a refactor and also risks accidentally breaking downstream.

Reply:
Really, it is only cubeviz that has the weird transposing due to specutils swapaxes. Everything else (like lcviz) should be in `else`.

Reply:
So, if we ever have TRampViz, it would theoretically follow the same indexing as lcviz.

Reply:
Does [1989b85](https://github.com/spacetelescope/jdaviz/pull/2678/commits/1989b85b830f1fd9c1548179975fe484d2406296) work for you?  Downstream this would then look something like https://github.com/spacetelescope/lcviz/pull/83